### PR TITLE
[Advanced Compiler]Fix upsample_linear1d

### PR DESCRIPTION
**Pull request by AdvancedCompiler**
### PR Category

Operator 
### Type of Change

Bug Fix
### Description

Fix the precision issue of upsample_linear1d op & supplement boundary cases in accuracy test.
### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [ ] Change is fully covered by a UT.

### Performance

cfd63d78d072eb9ddbde9a2b2eb83e92
ac4ea82122ca21769a77df4eb229813a
af591d8b19d71bb77d54560e8c07463b

### CODE REVIEW

File: src/flag_gems/ops/upsample_linear1d.py [hunk 1]
```diff
@@ -9,15 +9,14 @@
 
 
 @triton.jit
-def upsample_linear1d_kernel(
+def upsample_linear1d_kernel_optimized(
```
Comment (tengqm) [inline src/flag_gems/ops/upsample_linear1d.py]: There is no need to change this name eventually, right?

File: src/flag_gems/ops/upsample_linear1d.py [hunk 2]
```diff
@@ -65,7 +59,7 @@ def upsample_linear1d(
     align_corners: bool,
     scales: float = None,
 ):
-    logger.debug("GEMS UPSAMPLE LINEAR1D")
+    logger.debug("GEMS UPSAMPLE LINEAR1D OPTIMIZED")
```
Comment (tengqm) [inline src/flag_gems/ops/upsample_linear1d.py:62]: an optimized operator is still an implementation of the same operator ... so please don't change this.

File: src/flag_gems/ops/upsample_linear1d.py [hunk 3]
```diff
@@ -84,24 +78,31 @@ def upsample_linear1d(
     out = torch.empty((NC, W_out), device=self.device, dtype=self.dtype)
 
     if align_corners:
-        scale_ac = (W_in - 1) / (W_out - 1) if W_out > 1 else 0.0
-        scale_nc = 0.0
+        if W_out > 1:
+            scale_val = float((W_in - 1.0) / (W_out - 1.0))
```
Comment (tengqm) [inline src/flag_gems/ops/upsample_linear1d.py]: this `float` is not necessary right?
The `1.0` in the expression will ensure that scale_val is a float.
